Resources/public/js/translation.js

```diff
@@ -244,36 +244,11 @@ const TranslationManager = (() => {
         let params = {};
         let saveButton = document.getElementById('saveButton-' + lexikTranslationId);
         let trElement = saveButton.closest('tr.content');
-        let tdElements = trElement.querySelectorAll('td');
-        let translationsElements = trElement.querySelectorAll('td span.locale');
-
-        let spanColumnText = Array.from(tdElements).map(function (td, index) {
-            if (index <= 2) {
-                let span = td.querySelector('span');
-                if (span) {
-                    let th = trElement.closest('table').querySelectorAll('th')[index];
-                    return {
-                        value: span.innerText.trim(),
-                        column: th ? '_' + th.id.split('-')[0] : null
-                    };
-                }
-                return null;
-            }
-        }).filter(Boolean);
-
-        let translationsTexts = Array.from(translationsElements).map(function (translationElement) {
-            return translationElement.innerText.trim() + '-' + translationElement.id;
-        });
-
-        translationsTexts.forEach(function (translationText) {
-            let parts = translationText.split('-');
-            let translation = parts[0];
-            let locale = parts[3];
-            params[locale] = translation;
-        });
-
-        spanColumnText.forEach(function (spanText) {
-            params[spanText.column] = spanText.value;
+        let tdElements = trElement.querySelectorAll('td[class^="col-"]');
```

Comment:
While this is shorter it doesn't read very easily. After a while you can make out that text is collected from td elements (which probably contain translations). Can you make it easier to follow somehow?

Reply:
It simplify the code as we don't need to lookup for the related column header ID as we already have all needed information in the tr. We just have to iterate on td that have a class that start with ``col-`` to construct the parameters.

![image](https://github.com/lexik/LexikTranslationBundle/assets/652505/6a5d5834-b03e-4b8d-8a09-2cfff8e55e51)

It seems rather easy to read for me. Easier than before for sure.